core/types/hashing.go

```diff
@@ -36,6 +38,21 @@ var encodeBufferPool = sync.Pool{
 	New: func() interface{} { return new(bytes.Buffer) },
 }
 
+// getPooledBuffer retrieves a buffer from the pool and creates a byte slice
+// of the requested size from it.
+//
+// The caller should return the Buffer object back into encodeBufferPool after use!
```

Comment:
Would be good to clarify the semantics for the byte slice too, afaiu it also must not be kept after return? 
Something like this? 
```suggestion
// The caller should return the Buffer object back into encodeBufferPool after use, and must release the slice!
```